echopype/convert/set_groups_ek60.py

```diff
@@ -658,26 +667,32 @@ def set_vendor(self) -> xr.Dataset:
         # Retrieve pulse length and sa correction
         config = self.parser_obj.config_datagram["transceivers"]
         freq = [v["frequency"] for v in config.values()]
+        ch_ids = list(self.parser_obj.config_datagram["transceivers"].keys())
+        channel = [
+            self.parser_obj.config_datagram["transceivers"][ch_seq].get("channel_id", np.nan)
+            for ch_seq in ch_ids
+        ]
         pulse_length = np.array([v["pulse_length_table"] for v in config.values()])
         gain = np.array([v["gain_table"] for v in config.values()])
         sa_correction = [v["sa_correction_table"] for v in config.values()]
```

Comment:
(this is not part of this review)
sigh, the multiple list comprehensions... something again pointing to some re-organization of the parser data structure, but not now!